app/content/views/registration.py

```diff
@@ -58,9 +77,66 @@ def create(self, request, *args, **kwargs):
         registration = super().perform_create(
             serializer, event=event, user=request.user
         )
+
+        # Create order if event is a paid event
+        # Check if we have access token
+        # If not fetch access token
+        # Inlcude payment link in Serializer
+
+        if event.is_paid_event:
+            access_token = os.environ.get("PAYMENT_ACCESS_TOKEN")
+            expires_at = os.environ.get("PAYMENT_ACCESS_TOKEN_EXPIRES_AT")
+            if not access_token or datetime.now() >= datetime.fromtimestamp(
+                int(expires_at)
+            ):
+                (expires_at, access_token) = get_new_access_token()
+                os.environ.update({"PAYMENT_ACCESS_TOKEN": access_token})
+                os.environ.update({"PAYMENT_ACCESS_TOKEN_EXPIRES_AT": str(expires_at)})
+
+            prev_orders = Order.objects.filter(event=event, user=request.user)
+            has_paid_order = False
+
+            for order in prev_orders:
+                if (
+                    order.status == OrderStatus.CAPTURE
+                    or order.status == OrderStatus.RESERVE
+                    or order.status == OrderStatus.SALE
+                ):
+                    has_paid_order = True
+                    break
+
+            if not has_paid_order:
+
+                paytime = event.paid_information.paytime
+
+                expire_date = datetime.now() + timedelta(
+                    hours=paytime.hour, minutes=paytime.minute, seconds=paytime.second
+                )
+
+                # Create Order
+                order_id = uuid.uuid4()
+                amount = int(event.paid_information.price * 100)
+                res = initiate_payment(amount, str(order_id), event.title, access_token)
+                payment_link = res["url"]
+                order = Order.objects.create(
+                    order_id=order_id,
+                    user=request.user,
+                    event=event,
+                    payment_link=payment_link,
+                    expire_date=expire_date,
+                )
+                order.save()
+
+                check_if_has_paid.apply_async(
+                    args=(order.order_id, registration.registration_id),
+                    countdown=(paytime.hour * 60 + paytime.minute) * 60
+                    + paytime.second,
+                )
```

Comment:
Refaktorere dette om til sin egen funksjon kanskje?

Reply:
Jepp, har laget en egen funksjon for det nå som ligger i en util mappe i content mappen.